### PR TITLE
Check Linear Regression target column data type

### PR DESCRIPTION
**Pull request by fareeha**
Linear Regression accepts 'int' and 'float' target values. Previously `check_target_column_type()` which checks the type of target values was called on the validation set. Validation set is only created for a dataset size 100 or larger. For cases when data with fewer than 100 examples was given, the check was not applied and target values of type string were being accepted and handled incorrectly. 

The check for target values type is now applied on the target column of the data passed in. Fixes https://github.com/apple/turicreate/issues/1298

### CODE REVIEW

File: src/unity/toolkits/supervised_learning/supervised_learning.cpp [hunk 1]
```diff
@@ -1050,9 +1052,6 @@ void supervised_learning_model_base::api_train(
     
     valid_y = validation_data.select_columns({target}).materialize_to_sframe();
 
-    
-    check_target_column_type(this->name(), valid_y);
```
Comment (nickjong) [inline src/unity/toolkits/supervised_learning/supervised_learning.cpp]: Should we no longer be checking the validation set too?

Comment (fareeha) [reply src/unity/toolkits/supervised_learning/supervised_learning.cpp]: Validation set is created from the dataset that is provided by the user. So I feel it might be repetitive to also call the check function again on a section of that dataset.

Comment (nickjong) [reply src/unity/toolkits/supervised_learning/supervised_learning.cpp]: Although validation data can be automatically sampled from the training data, I believe our API also allows users to provide the validation set as a completely unrelated SFrame.

Comment (fareeha) [reply src/unity/toolkits/supervised_learning/supervised_learning.cpp]: Yes, you're right. Thank you for pointing that! It was worth verifying that case as well. So, in the case when a user provides their own validation set: the model `create()` calls `_validate_data(dataset, validation_set)`. The validation_set is set to 'auto' if we want the validation set to the generated automatically or it can be given an SFrame containing the validation set. In the latter case, `_validate_data(dataset, validation_set)` applies append on the two SFrames using `validation_set.head().append(dataset.head())`  to check schema. This allows verifying that the columns and their types of both `dataset` and `validation_data` match.
Since we are now checking the type of the `dataset` target column, and by extension we make sure that the `validation_data` also has the matching valid column type.

Comment (hoytak) [reply src/unity/toolkits/supervised_learning/supervised_learning.cpp]: I think the check should probably be in both places.  That's the easiest fix here.

Comment (fareeha) [reply src/unity/toolkits/supervised_learning/supervised_learning.cpp]: Good idea, to be safe the check on validation data column type should be restored.
